File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int maxprofit = 0;

        for(int i = 1; i < prices.length; i++){
            if(prices[i] < prices[left]){
                left = i;
            }else{
                int currprofit = prices[i] - prices[left];
                maxprofit = Math.max(currprofit,maxprofit);
            }

        }
        return maxprofit;


        
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:


        maxprofit = 0
        left = 0

      


      

        for i in range(len(prices)):

            if prices[i] < prices[left]:
                left = i

            else:
                currprofit = prices[i] - prices[left]
                maxprofit = max(maxprofit, currprofit)


        return maxprofit



